server.go

```diff
@@ -620,14 +620,11 @@ func (res *responseWriter) waitReadyWrite() {
 
 var (
 	// ErrServerClosed is returned by Server.Serve when the server is closed.
-	ErrServerClosed = errors.New("redis: Server closed")
-)
-
-var (
-	errNegativeStreamCount       = errors.New("invalid call to redis.ResponseWriter.Stream with a negative value")
-	errStreamCalledAfterWrite    = errors.New("invalid call to redis.ResponseWriter.Stream after redis.ResponseWriter.Write was called")
-	errStreamCalledTooManyTimes  = errors.New("multiple calls to ResponseWriter.Stream")
-	errWriteCalledTooManyTimes   = errors.New("too many calls to redis.ResponseWriter.Write")
-	errWriteCalledNotEnoughTimes = errors.New("not enough calls to redis.ResponseWriter.Write")
-	errHijacked                  = errors.New("invalid use of a hijacked redis.ResponseWriter")
+	ErrServerClosed              = errors.New("redis: Server closed")
+	ErrNegativeStreamCount       = errors.New("invalid call to redis.ResponseWriter.Stream with a negative value")
+	ErrStreamCalledAfterWrite    = errors.New("invalid call to redis.ResponseWriter.Stream after redis.ResponseWriter.Write was called")
+	ErrStreamCalledTooManyTimes  = errors.New("multiple calls to ResponseWriter.Stream")
+	ErrWriteCalledTooManyTimes   = errors.New("too many calls to redis.ResponseWriter.Write")
+	ErrWriteCalledNotEnoughTimes = errors.New("not enough calls to redis.ResponseWriter.Write")
+	ErrHijacked                  = errors.New("invalid use of a hijacked redis.ResponseWriter")
```

Comment:
I made those errors public for two reasons:
- I often find it really frustrating when libraries don't expose their error types, it makes it hard to take appropriate behaviors based on the specific errors that they generate
- I moved the server_test.go file from being in the `redis` package to the `redis_test` package, so we can better reuse test helpers between test files, therefore the unexported symbols couldn't be accessed anymore